workflow/parse_nextflow.py

```diff
@@ -150,28 +150,66 @@ def parse_method_versions(results_path, results):
     return results
 
 
-def compute_ranking(task_name, dataset_results):
-    """Rank all methods on a specific dataset."""
-    rankings = np.zeros(len(dataset_results))
+def normalize_scores(task_name, dataset_results):
+    """Normalize method scores to [0, 1] based on baseline method scores."""
     metric_names = list(dataset_results.values())[0]["metrics"].keys()
     for metric_name in metric_names:
         metric = openproblems.api.utils.get_function(task_name, "metrics", metric_name)
-        sorted_order = np.argsort(
+        metric_scores = np.array(
             [
                 dataset_results[method_name]["metrics"][metric_name]
                 for method_name in dataset_results
             ]
         )
-        if metric.metadata["maximize"]:
-            sorted_order = sorted_order[::-1]
-        rankings += np.argsort(sorted_order)
+        baseline_methods = [
+            method_name
+            for method_name in dataset_results
+            if openproblems.api.utils.get_function(
+                task_name, "methods", method_name
+            ).metadata["is_baseline"]
+        ]
+        if len(baseline_methods) < 2:
+            # just use all methods as a fallback
+            baseline_methods = dataset_results.keys()
+        baseline_scores = np.array(
+            [
+                dataset_results[method_name]["metrics"][metric_name]
+                for method_name in baseline_methods
+            ]
+        )
+        metric_scores -= baseline_scores.min()
+        metric_scores /= baseline_scores.max()
+        if not metric.metadata["maximize"]:
+            metric_scores = 1 - metric_scores
+        for method_name, score in zip(dataset_results, metric_scores):
+            dataset_results[method_name]["metrics"][metric_name] = score
+    return dataset_results
+
+
+def drop_baselines(task_name, dataset_results):
+    """Remove baseline methods from dataset results."""
+    for method_name in dataset_results.keys():
+        method = openproblems.api.utils.get_function(task_name, "methods", method_name)
+        if method.metadata["is_baseline"]:
+            del dataset_results[method_name]
+    return dataset_results
```

Comment:
great! That answers my question about not dropping non-baseline methods even if they are used as baseline